### DISCUSSION TITLE
Using OSC 133 sequences to select last block of command output, in Zsh?

### DISCUSSION THREAD

**Discussion by AndydeCleyre** [tmux/tmux#4128] [General]
Hello! I'd like to bind a key to enter copy mode and select the last command's output, in Zsh. I feel close but am not yet successful.

I'm thinking the binding can be done either within Zsh or the tmux conf, and I'm happy with either, but my current attempt is within Zsh.

---

What I'm trying:

Within the tmux conf, I `set -g allow-passthrough on`.

To ensure the escape sequences exist to be found, in Zsh:

```zsh
autoload -Uz add-zsh-hook

.zshrc_prompt-mark-output-start () {
  print -n '\e]133;C\a'
}

add-zsh-hook preexec .zshrc_prompt-mark-output-start

PROMPT=$'\e]133;D\a'$'\n''${(j: :)psvar}'$'\n''%B%F{white}-- %F{green}%(!.#.$)%f%b '
```

To bind the key, in Zsh:

```zsh
.zle_tmux-copy-mode-last-output () {
  tmux \
    copy-mode \; \
    send -X search-backward $'\e]133;D' \; \
    send -X begin-selection \; \
    send -X search-backward $'\e]133;C' \; \
    send -X stop-selection
}
zle -N            .zle_tmux-copy-mode-last-output
bindkey '^[[5;2~' .zle_tmux-copy-mode-last-output  # shift+pgup
```

The problem *may* just be improper formatting of the escape sequences to find, or it may be more than that. I've been trying some variants of the escape sequence formatting, including e.g. `$'\033]133;D'` and `'\033]133;D'`.

Any pointers toward this goal would be much appreciated!

**Comment by nicm** [comment]
You can't search for escape sequences, you'll need to make the prompt contain something visible, or use next-prompt and previous-prompt if your tmux is new enough.
